compiler/rustc_trait_selection/src/solve/project_goals.rs

```diff
@@ -40,7 +40,7 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
                 }
             }
             ty::AliasKind::Opaque => self.normalize_opaque_type(goal),
-            ty::AliasKind::Inherent => bug!("IATs not supported here yet"),
+            ty::AliasKind::Weak | ty::AliasKind::Inherent => bug!("IATs not supported here yet"),
```

Comment:
this should be fixed by adding `normalize_weak_type` that just calls type_of and register's the type alias's predicates.

Reply:
I'm not sure how to hit this code path with weak type aliases. Would we ever end up with a projection predicate for weak type aliases at all?

Reply:
via assemble_candidates_after_normalizing_self_type, i would assume

Reply:
This code path is now tested in a preexisting (since a few days) test: [tests/ui/traits/new-solver/dont-remap-tait-substs.rs](https://github.com/rust-lang/rust/blob/3d09b990d7e3d8539a3b9655894e7e05dcc9ebf2/tests/ui/traits/new-solver/dont-remap-tait-substs.rs)